kernel/src/grouping_iterators/version_grouping_iterator.rs

```diff
@@ -30,33 +30,35 @@ impl<L: AsUrl> VersionGroup<L> {
 // I: The concrete iterator type that yields ParsedLogPath<L>
 // L: The type implementing AsUrl that represents the underlying file location
 // This allows for flexible iteration over log files while maintaining type safety
-pub struct VersionGroupingIterator<I, L> 
-where 
+pub struct VersionGroupingIterator<I, L>
+where
     L: AsUrl,
     I: Iterator<Item = ParsedLogPath<L>>,
 {
     files: Peekable<I>,
 }
 
 // We use a type conversion to allow the caller to pass any iterator that yields ParsedLogPath
-// This gives an advantage to group files by version in a streaming fashion if we can assume that 
+// This gives an advantage to group files by version in a streaming fashion if we can assume that
 // the input iterator is already sorted by version, like an S3 listing of delta log files.
-impl<I, L> From<I> for VersionGroupingIterator<I,L>
+impl<I, L> From<I> for VersionGroupingIterator<I, L>
 where
     L: AsUrl,
     I: Iterator<Item = ParsedLogPath<L>>,
 {
     fn from(value: I) -> Self {
-        let files  = value;
-        VersionGroupingIterator { files: files.peekable() }
+        let files = value;
+        VersionGroupingIterator {
+            files: files.peekable(),
+        }
```

Comment:
`cargo fmt` will force each value to its own line unless they're simple identifiers, so it's worth pulling everything out to the helper variable:
```suggestion
        let files = value.peekable();
        VersionGroupingIterator { files }
```